spikeinterface/preprocessing/average_across_direction.py

```diff
@@ -0,0 +1,154 @@
+import numpy as np
+from spikeinterface.core import BaseRecording, BaseRecordingSegment
+from .basepreprocessor import BasePreprocessorSegment
+from spikeinterface.core.core_tools import define_function_from_class
+
+
+class AverageAcrossDirectionRecording(BaseRecording):
+    name = "average_across_direction"
+    installed = True
+
+    def __init__(
+        self,
+        parent_recording: BaseRecording,
+        direction: str = "y",
+        dtype=None,
+    ):
+        """Averages channels at the same position along `direction
+
+        The resulting recording will live on virtual channels located
+        at each *sorted* unique position along direction, and with other
+        direction's positions set to 0.
+
+        Parameters
+        ----------
+        parent_recording : BaseRecording
+            recording to zero-pad
+        direction : str
+            Channels living at unique positions along this direction
+            will be averaged.
+        dtype : optional numpy dtype
+            If supplied, convert recording to this type before averaging.
+            If unset, if the recording is not a floating type, it is converted
+            to float32 by default, or kept as its floating type otherwise.
```

Comment:
In almost all preprocessing the most often behavior is to keep the parents dtype when it is None.
What is the deep underlying reaon for forcing to float ?

Reply:
To clarify, numpy _calculations_ promote to `float` where it makes sense to do so, but numpy _array storage_ is always fixed in its allocation to disk and will therefore always _overflow_ for `int` types

As I read the docstring (though I could be very wrong), the `dtype` refers here to the storage pattern of the time series recording, right?

Reply:
There are two dtypes involved here. The first is the dtype of the parent recording (before the preprocessing), and the second is the dtype after preprocessing. The parent dtype could be an integer type, and as you say, numpy will promote this when calculating the mean to a float type. This preprocessor does the same, so the output of the preprocessor in the case of a int parent recording and when passing `dtype=None` will be a recording with float32 type, extracted on the fly from the underlying parent recording.

Reply:
Hmm... I see now

Examples go a long way to conveying meaning

And specific explanation of *why* you chose to do it that particular way (maintaining floating point accuracy vs. saving memory space vs. speed of processing) would also go a long way